core/arch/aarch64/instr_create.h

```diff
@@ -369,8 +369,11 @@
  * they just need to know whether they need to preserve the app's flags, so maybe
  * we can just document that this may not write them.
  */
-#define XINST_CREATE_slr_s(dc, d, s) \
-  instr_create_1dst_2src((dc), OP_lsrv, (d), (d), (s))
+#define XINST_CREATE_slr_s(dc, d, rm_or_imm) \
+  opnd_is_reg(rm_or_imm) ? \
+    instr_create_1dst_2src((dc), OP_lsrv, (d), (d), (rm_or_imm)) : \
+    instr_create_1dst_3src((dc), OP_ubfm, (d), (d), (rm_or_imm), \
```

Comment:
So the FIXME above still applies as these are using raw creation rather than INSTR_CREATE macros but the comment should be updated to include INSTR_CREATE_ubfm.